src/util/connections.ts

```diff
@@ -92,14 +92,16 @@ export async function addConnection(clusterConnectionTreeProvider: ClusterConnec
   currentPanel.webview.onDidReceiveMessage(async (message: any) => {
     switch (message.command) {
       case 'submit':
-        let url;
+        const url = message.isSecure ? (Constants.prefixSecureURL + message.url) : (Constants.prefixURL + message.url);
         try {
-          if (message.isSecure) {
-            url = Constants.prefixSecureURL + message.url;
-          } else {
-            url = Constants.prefixURL + message.url;
-          }
-          await Cluster.connect(url, { username: message.username, password: message.password, configProfile: 'wanDevelopment' });
+          await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: 'Checking connection...',
+            cancellable: true
+          }, async () => {
+            await Cluster.connect(url, { username: message.username, password: message.password, configProfile: 'wanDevelopment' });
```

Comment:
Have you tested it with invalid credentials as well? Because the "connect" in couchbase is often lazy, so nothing happens until you actually request something from the server.
In most SDKS the above call won't throw an error even with invalid credentials.

Reply:
@deniswsrosa, yes I have tested it with wrong credentials and it throws error for me each time.

Reply:
Ok!
thank you!